File: mockez-front/src/app/core/service/group.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '@environment/environment';
import { Group } from '@core/model/group';

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

@Injectable({
  providedIn: 'root'
})
export class GroupService {

  constructor(
    private httpClient: HttpClient
  ) {
  }

  getGroupsWithAccess(): Observable<Group[]> {
    return this.httpClient.get<Group[]>(`${environment.apiUrl}/group`);
  }

  getGroup(id: string): Observable<Group> {
    return this.httpClient.get<Group>(`${environment.apiUrl}/group/${id}`);
  }

  saveOrUpdate(group: Group): Observable<string> {
    return this.httpClient.post<string>(`${environment.apiUrl}/group`, group);
  }

  delete(id: string): Observable<string> {
    return this.httpClient.delete<string>(`${environment.apiUrl}/group/${id}`);
  }
}

File: mockez-back/src/main/java/com/mockez/service/GroupService.java
package com.mockez.service;

import com.mockez.domain.model.entity.Group;

import java.util.List;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

public interface GroupService {

    List<Group> getGroupsWithAccess(UUID userId);

    Group getProject(UUID id);

    UUID delete(UUID id);

    UUID saveOrUpdate(Group group);
}
